pkg/backend/policypack.go

```diff
@@ -31,8 +31,8 @@ type PublishOperation struct {
 	Scopes     CancellationScopeSource
 }
 
-// ApplyOperation publishes a PolicyPack to the backend.
-type ApplyOperation struct {
+// PolicyPackOperation is used to make various operations against a Policy Pack.
+type PolicyPackOperation struct {
```

Comment:
I'll admit I don't quite understand how this is used.

If I have a `PolicyPack` instance, and want to, e.g. `Publish` it, the `PolicyPackOperation` contains the details about what specific version of the policy pack I want to publish? (As well as the cancellation source.)

Would it make more sense to have the `PolicyPack` instance be specific to a version instead? e.g. the interface has a `Version() int` property?

If you are happy with the design, feel free to keep it as-is. It just doesn't quite look right at the first glance, IMHO.

Reply:
I agree this design was a bit confusing, I was just sort of going off whats already there. I'll play around and see if I can make any improvements here. May choose to do it in a later PR just to simplify things